src/services/earn/earn-service.ts

```diff
@@ -22,15 +30,52 @@ import { IProviderService } from '@services/providers';
 import { MULTICALL_CONTRACT } from '@services/providers/utils';
 import { IAllowanceService } from '@services/allowances';
 import { PERMIT2_CONTRACT } from '@services/permit2/utils/config';
+import qs from 'qs';
+import { IFetchService } from '@services/fetch';
 
 export class EarnService implements IEarnService {
   constructor(
+    private readonly apiUrl: string,
     private readonly permit2Service: IPermit2Service,
     private readonly quoteService: IQuoteService,
     private readonly providerService: IProviderService,
-    private readonly allowanceService: IAllowanceService
+    private readonly allowanceService: IAllowanceService,
+    private readonly fetchService: IFetchService
   ) {}
 
+  async getSupportedStrategies(args?: { chains?: ChainId[]; config?: { timeout: TimeString } }) {
+    const params = qs.stringify({ chains: args?.chains }, { arrayFormat: 'comma', skipNulls: true });
+    const url = `${this.apiUrl}/v1/earn/strategies/supported?${params}`;
+    const response = await this.fetchService.fetch(url, { timeout: args?.config?.timeout });
+    const body: SupportedStrategiesResponse = await response.json();
+    const result: Record<ChainId, Strategy[]> = {};
+    Object.entries(body.strategiesByNetwork).forEach(([stringChainId, { strategies, tokens }]) => {
+      const chainId = Number(stringChainId) as ChainId;
+      result[chainId] = strategies.map(({ farm: { rewards, asset, ...restFarm }, guardian, ...strategyResponse }) => {
+        const strategy: Strategy = {
+          ...{
+            ...strategyResponse,
+            farm: {
+              ...restFarm,
+              asset: tokens[asset],
+            },
+          },
+        };
+        if (rewards) {
+          strategy.farm.rewards = {
+            tokens: rewards ? rewards?.tokens.map((token) => tokens[token]) : [],
```

Comment:
I think you already checked this
```suggestion
            tokens: rewards.tokens.map((token) => tokens[token]),
```